homeassistant/components/vacuum/xiaomi.py

```diff
@@ -0,0 +1,375 @@
+"""
+Support for the Xiaomi vacuum cleaner robot.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/vacuum.xiaomi_vacuum/
+"""
+import asyncio
+from functools import partial
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant.components.vacuum import (
+    VacuumDevice, DOMAIN,
+    DEFAULT_ICON, PLATFORM_SCHEMA, VACUUM_SERVICE_SCHEMA,
+    SUPPORT_TURN_ON, SUPPORT_TURN_OFF, SUPPORT_PAUSE, SUPPORT_RETURN_HOME,
+    SUPPORT_STOP, SUPPORT_LOCATE, SUPPORT_STATUS, SUPPORT_BATTERY,
+    SUPPORT_FANSPEED, SUPPORT_SENDCOMMAND)
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import (
+    STATE_ON, STATE_OFF, ATTR_ENTITY_ID, CONF_NAME, CONF_HOST, CONF_TOKEN)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['python-mirobo==0.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Xiaomi Vacuum cleaner'
+ICON = DEFAULT_ICON
+PLATFORM = 'xiaomi'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_TOKEN): vol.All(str, vol.Length(min=32, max=32)),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+}, extra=vol.ALLOW_EXTRA)
+
+SERVICE_MOVE_REMOTE_CONTROL = 'xiaomi_remote_control_move'
+SERVICE_MOVE_REMOTE_CONTROL_STEP = 'xiaomi_remote_control_move_step'
+SERVICE_START_REMOTE_CONTROL = 'xiaomi_remote_control_start'
+SERVICE_STOP_REMOTE_CONTROL = 'xiaomi_remote_control_stop'
+
+SPEED_LOW = ('Quiet', 38)
+SPEED_MEDIUM = ('Balanced', 60)
+SPEED_HIGH = ('Turbo', 77)
+SPEED_MAX = ('Max', 90)
+FANSPEED_LABELS, FANSPEED_VALUES = list(
+    zip(*[SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH, SPEED_MAX]))
+
+ATTR_RC_VELOCITY = 'velocity'
+ATTR_RC_ROTATION = 'rotation'
+ATTR_RC_DURATION = 'duration'
+
+SERVICE_SCHEMA_REMOTE_CONTROL = VACUUM_SERVICE_SCHEMA.extend({
+    vol.Optional(ATTR_RC_VELOCITY):
+        vol.All(vol.Coerce(float), vol.Range(min=-0.29, max=0.29)),
+    vol.Optional(ATTR_RC_ROTATION):
+        vol.All(vol.Coerce(int), vol.Range(min=-179, max=179)),
+    vol.Optional(ATTR_RC_DURATION): cv.positive_int,
+})
+
+SERVICE_TO_METHOD = {
+    SERVICE_START_REMOTE_CONTROL: {'method': 'async_start_remote_control'},
+    SERVICE_STOP_REMOTE_CONTROL: {'method': 'async_stop_remote_control'},
+    SERVICE_MOVE_REMOTE_CONTROL: {
+        'method': 'async_move_remote_control',
+        'schema': SERVICE_SCHEMA_REMOTE_CONTROL},
+    SERVICE_MOVE_REMOTE_CONTROL_STEP: {
+        'method': 'async_move_remote_control_step',
+        'schema': SERVICE_SCHEMA_REMOTE_CONTROL},
+}
```

Comment:
Considering the keys here are used only in this place, in my opinion it would enhance readability to not to define these above as globals but just use them directly here.. Opinions regarding to this may vary though..